streams/src/test/java/org/apache/kafka/streams/integration/utils/KafkaEmbedded.java

```diff
@@ -171,24 +177,43 @@ public void createTopic(final String topic,
                             final Properties topicConfig) {
         log.debug("Creating topic { name: {}, partitions: {}, replication: {}, config: {} }",
             topic, partitions, replication, topicConfig);
-        try (KafkaZkClient kafkaZkClient = createZkClient()) {
-            final AdminZkClient adminZkClient = new AdminZkClient(kafkaZkClient);
-            adminZkClient.createTopic(topic, partitions, replication, topicConfig, RackAwareMode.Enforced$.MODULE$);
+        final NewTopic newTopic = new NewTopic(topic, partitions, (short) replication);
+        newTopic.configs((Map) topicConfig);
+
+        try (final AdminClient adminClient = createAdminClient()) {
+            adminClient.createTopics(Collections.singletonList(newTopic)).all().get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException(e);
         }
     }
 
-    private KafkaZkClient createZkClient() {
-        return KafkaZkClient.apply(zookeeperConnect(), false, DEFAULT_ZK_SESSION_TIMEOUT_MS,
-                DEFAULT_ZK_CONNECTION_TIMEOUT_MS, Integer.MAX_VALUE, Time.SYSTEM, "testMetricGroup", "testMetricType");
+    public AdminClient createAdminClient() {
+        final Properties adminClientConfig = new Properties();
+        adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList());
+        final Object listeners = effectiveConfig.get(KafkaConfig$.MODULE$.ListenersProp());
+        if (listeners != null && listeners.toString().contains("SSL")) {
+            adminClientConfig.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, effectiveConfig.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG));
+            adminClientConfig.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, ((Password) effectiveConfig.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG)).value());
+            adminClientConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL");
+        }
+        return AdminClient.create(adminClientConfig);
     }
 
     public void deleteTopic(final String topic) {
         log.debug("Deleting topic { name: {} }", topic);
+        try (final AdminClient adminClient = createAdminClient()) {
+            adminClient.deleteTopics(Collections.singletonList(topic)).all().get();
+        } catch (InterruptedException | ExecutionException e) {
+            if (!(e.getCause() instanceof UnknownTopicOrPartitionException))
```

Comment:
Might be better to add a catch-clause for this case instead?
```
try {
...
} catch (final UnknownTopicOrPartitionException ignoreAndSwallow) {
} catch (final InterruptedException | ExecutionException e) {
    throw new RuntimeException(e);
}
```

Reply:
`UnknownTopicOrPartitionException` is the cause of the actual exception `e`, so we cannot just catch it here.

Reply:
Makes sense.